src/os/bluestore/BlueStore.cc

```diff
@@ -3228,8 +3249,108 @@ void BlueStore::ExtentMap::make_range_shared_maybe_merge(
     txc->write_onode(oldo);
   }
 }
+#endif
 
-#ifdef WITH_ESB
+#ifndef WITH_ESB
```

Comment:
IMO dup() method is the only major location where the existing and new  functionalities diverge. Most of other code changes do not impact existing functionality (with some minor exceptions which I believe we can accept without ifdefs) but mostly introduce new methods for the new one. Given that wouldn't be better to introduce runtime parameter to enable/disable the new dup logic and get rid off ifdef-s? IMO this would be much more flexible from user'ss POV  and the code will remain clean...

Reply:
ifdefs gone, config option in.